lightning/src/util/wakers.rs

```diff
@@ -175,6 +119,9 @@ impl FutureState {
 }
 
 /// A simple future which can complete once, and calls some callback(s) when it does so.
+///
+/// Clones can be made and all futures cloned from the same source will complete at the same time.
+#[derive(Clone)]
 pub struct Future {
```

Comment:
a bit confusing to see `Future` != `StdFuture`. maybe rename to `Notified` or `NotifiedFuture`?